### ISSUE TITLE
CGAL doesn't automatically pull in mpir and mpfr

### ISSUE THREAD

**Issue by rkcarroll** [microsoft/vcpkg#5662]
I had to manually add linker references to the vcpkg MPIRand MPFR libraries in order to successfully link a program that uses CGAL

**Comment by wangli28** [comment]
Hi @rkcarroll,
Can you provide more information? How did you build this library? And what information did vcpkg gives to you?
Thanks.

**Comment by acgetchell** [comment]
I'd also like to know!

**Comment by rkcarroll** [comment]
Honestly, it's been a few weeks since I made the original post, so I can't guarantee the accuracy of this statement, but here is what I remember:  CGAL references MPIR and MPFR; however, installing vcpkg's CGAL package does not automatically install the vcpkg MPIR's and MPFR's packages.  I don't think the Linux version of CGAL requires these packages, btw.  Anyway, manually installing the vcpkg MPIR and MPFR fixes the problem.

**Comment by acgetchell** [comment]
Ah, I've tried pre-installing `mpir`(which works) and `mpfr` (which doesn't -- see https://github.com/Microsoft/vcpkg/issues/5910).

**Comment by wangli28** [comment]
I installed CGAL successfully with VS2017 15.9.11 and VCPKG latest source on Windows 10. Its dependencies contain MPIR and MPFR. This is my build log.

> PS F:\vcpkg> .\vcpkg.exe install cgal
The following packages will be built and installed:
 / * boost-any[core]:x86-windows
 / * boost-bimap[core]:x86-windows
......
/    cgal[core]:x86-windows
/  * liblzma[core]:x86-windows
/  * mpfr[core]:x86-windows
/  * mpir[core]:x86-windows
/  * zstd[core]:x86-windows
Additional packages (*) will be modified to complete this operation.
Starting package 1/32: mpir:x86-windows
Building package mpir[core]:x86-windows...
-- Downloading https://github.com/wbhart/mpir/archive/mpir-3.0.0.tar.gz...
-- Extracting source F:/vcpkg/downloads/wbhart-mpir-mpir-3.0.0.tar.gz
-- Applying patch enable-runtimelibrary-toggle.patch
-- Using source at F:/vcpkg/buildtrees/mpir/src/mpir-3.0.0-c9f63a8302
-- Building F:/vcpkg/buildtrees/mpir/x86-windows/build.vc15/dll_mpir_gc/dll_mpir_gc.vcxproj for Release
-- Building F:/vcpkg/buildtrees/mpir/x86-windows/build.vc15/dll_mpir_gc/dll_mpir_gc.vcxproj for Debug
-- Performing post-build validation
-- Performing post-build validation done
Building package mpir[core]:x86-windows... done
Installing package mpir[core]:x86-windows...
Installing package mpir[core]:x86-windows... done
Elapsed time for package mpir:x86-windows: 1.788 min
Starting package 2/32: mpfr:x86-windows
Building package mpfr[core]:x86-windows...
-- Downloading http://www.mpfr.org/mpfr-4.0.1/mpfr-4.0.1.tar.xz...
-- Extracting source F:/vcpkg/downloads/mpfr-4.0.1.tar.xz
-- Configuring x86-windows
-- Building x86-windows-dbg
-- Building x86-windows-rel
-- Performing post-build validation
-- Performing post-build validation done
Building package mpfr[core]:x86-windows... done
Installing package mpfr[core]:x86-windows...
Installing package mpfr[core]:x86-windows... done
Elapsed time for package mpfr:x86-windows: 30.64 s
Starting package 3/32: boost-any:x86-windows
......
Building package cgal[core]:x86-windows... done
Installing package cgal[core]:x86-windows...
Installing package cgal[core]:x86-windows... done
Elapsed time for package cgal:x86-windows: 1.914 min
/
Total elapsed time: 10.14 min
/
The package cgal:x86-windows provides CMake targets:
/
    find_package(CGAL CONFIG REQUIRED)
    # Note: 1 target(s) were omitted.
    target_link_libraries(main PRIVATE ${cgal_lib} CGAL::${cgal_lib} CGAL_Qt5_moc_and_resources CGAL::Qt5_moc_and_resources)

**Comment by lrineau** [comment]
@Rastaban I think this issue can be closed. It is fixed.

**Comment by acgetchell** [comment]
It's not resolved on all platforms:

macOS 10.14:
```
┌─[adam][hapkido][±][master ✓][~/vcpkg]
└─▪ ./vcpkg install cgal
The following packages will be built and installed:
    cgal[core]:x64-osx
  * mpfr[core]:x64-osx
Additional packages (*) will be modified to complete this operation.
Starting package 1/2: mpfr:x64-osx
Building package mpfr[core]:x64-osx... 
-- Using cached /Users/adam/vcpkg/downloads/mpfr-4.0.1.tar.xz
-- Configuring x64-osx-dbg
-- Configuring x64-osx-rel
-- Building x64-osx-dbg
CMake Error at scripts/cmake/vcpkg_build_cmake.cmake:202 (message):
    Command failed: /usr/local/Cellar/cmake/3.14.3/bin/cmake;--build;.;--config;Debug;--target;install;--;-v
    Working Directory: /Users/adam/vcpkg/buildtrees/mpfr/x64-osx-dbg
    See logs for more information:
      /Users/adam/vcpkg/buildtrees/mpfr/install-x64-osx-dbg-out.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_install_cmake.cmake:24 (vcpkg_build_cmake)
  ports/mpfr/portfile.cmake:18 (vcpkg_install_cmake)
  scripts/ports.cmake:71 (include)


Error: Building package mpfr:x64-osx failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: mpfr:x64-osx
  Vcpkg version: 2018.11.23-unknownhash

Additionally, attach any relevant sections from the log files above.
```

**Comment by oxydated** [comment]
I'm having similar issues with Windows 10:

```
The following packages will be built and installed:
    mpfr[core]:x86-windows
Starting package 1/1: mpfr:x86-windows
Building package mpfr[core]:x86-windows...
Warning: The following VS instances are excluded because the English language pack is unavailable.
    C:\Program Files (x86)\Microsoft Visual Studio\2019\Community
    C:\Program Files (x86)\Microsoft Visual Studio\2017\Community
Please install the English language pack.
-- Downloading http://www.mpfr.org/mpfr-4.0.2/mpfr-4.0.2.tar.xz...
-- Downloading http://www.mpfr.org/mpfr-4.0.2/mpfr-4.0.2.tar.xz... Failed. Status: 7;"Couldn't connect to server"
CMake Error at scripts/cmake/vcpkg_download_distfile.cmake:175 (message):

      Failed to download file.
      If you use a proxy, please set the HTTPS_PROXY and HTTP_PROXY environment
      variables to "https://user:password@your-proxy-ip-address:port/".
      Otherwise, please submit an issue at https://github.com/Microsoft/vcpkg/issues

Call Stack (most recent call first):
  ports/mpfr/portfile.cmake:3 (vcpkg_download_distfile)
  scripts/ports.cmake:94 (include)
```
It seems like the http link should be https.

**Comment by JackBoosY** [comment]
@oxydated I think you can't access the download link. Can I manually download mpfr in [this](http://www.mpfr.org/mpfr-4.0.2/mpfr-4.0.2.tar.xz) link?

**Comment by oxydated** [comment]
> @oxydated I think you can't access the download link. Can I manually download mpfr in [this](http://www.mpfr.org/mpfr-4.0.2/mpfr-4.0.2.tar.xz) link?

Yes, I can.

**Comment by JackBoosY** [comment]
@oxydated Okay, I confirm we should use `https` instead of `http`, we will fix it later.

**Comment by lrineau** [comment]
I have tried `vcpkg install cgal` on Linux, and `mpir` and `mpfr` were pulled in.

**Comment by JackBoosY** [comment]
This issue should be fixed in #8659.

**Comment by JackBoosY** [comment]
@oxydated Please download mpfr manually by this [link](http://www.mpfr.org/mpfr-4.0.2/mpfr-4.0.2.tar.xz) and put it into _VCPKG_PATH/downloads/_, then rebuild mpfr.

**Comment by JackBoosY** [comment]
Hi everyone, this issue should be fixed in #8659.
Please update vcpkg and rebuild cgal.

Thanks.
